crates/rover-client/src/operations/persisted_queries/publish/types.rs

```diff
@@ -107,6 +113,145 @@ impl From<PersistedQueriesPublishInput> for QueryVariables {
     }
 }
 
+#[derive(Debug, Serialize, Deserialize)]
+pub struct RelayPersistedQueryManifest {
+    #[serde(flatten)]
+    operations: HashMap<String, String>,
+}
+
+impl TryFrom<RelayPersistedQueryManifest> for ApolloPersistedQueryManifest {
+    type Error = RoverClientError;
+
+    fn try_from(relay_manifest: RelayPersistedQueryManifest) -> Result<Self, Self::Error> {
+        let mut anonymous_operations = Vec::new();
+        let mut syntax_errors = Vec::new();
+        let mut ids_with_multiple_operations = Vec::new();
+        let mut ids_with_no_operations = Vec::new();
+        let mut operations = Vec::new();
+        for (id, body) in relay_manifest.operations {
+            let ast = Parser::new(&body).parse();
+
+            let operation_definitions: Vec<OperationDefinition> = ast
+                .clone()
+                .document()
+                .definitions()
+                .filter_map(|definition| {
+                    if let Definition::OperationDefinition(operation_definition) = definition {
+                        Some(operation_definition)
+                    } else {
+                        None
+                    }
+                })
+                .collect();
+
+            let maybe_definition = match operation_definitions.len() {
+                0 => {
+                    ids_with_no_operations.push(id.clone());
+                    None
+                }
+                1 => {
+                    if let Some(operation_definition) = operation_definitions.get(0) {
```

Comment:
How can this `if` ever fail?
(Maybe the match could change to some sort of match against a slice instead so we're directly pulling out operation_definition, instead of using `get` and introducing an `else` clause that can't happen? https://stackoverflow.com/questions/36262937/how-can-i-pattern-match-on-a-vector-of-string-slices-and-extract-values )